src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpOnFilenameCheck.java

```
@@ -0,0 +1,179 @@
+    private boolean ignoreFileExtensions;
+
+    /**
+     * Setter for folder format.
+     *
+     * @param folderPattern format of folder.
+     * @throws org.apache.commons.beanutils.ConversionException if unable to
+     *         create Pattern object.
+     */
+    public void setFolderPattern(String folderPattern) {
+        this.folderPattern = CommonUtils.createPattern(folderPattern);
+    }
+
+    /**
+     * Setter for file name format.
+     *
+     * @param fileNamePattern format of file.
+     * @throws org.apache.commons.beanutils.ConversionException if unable to
+     *         create Pattern object.
+     */
+    public void setFileNamePattern(String fileNamePattern) {
+        this.fileNamePattern = CommonUtils.createPattern(fileNamePattern);
+    }
+
+    /**
+     * Sets whether the check should look for a file name match or mismatch.
+     *
+     * @param match check's option for matching file names.
+     */
+    public void setMatch(boolean match) {
+        this.match = match;
+    }
+
+    /**
+     * Sets whether file name matching should drop the file extension or not.
+     *
+     * @param ignoreFileExtensions check's option for ignoring file extension.
+     */
+    public void setIgnoreFileExtensions(boolean ignoreFileExtensions) {
+        this.ignoreFileExtensions = ignoreFileExtensions;
+    }
+
+    @Override
+    public void init() {
+        if (fileNamePattern == null && folderPattern == null) {
+            fileNamePattern = CommonUtils.createPattern("\\s");
+            match = true;
+        }
+    }
+
+    @Override
+    protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
+        String filePath;
+        String folderPath;
+
+        try {
+            filePath = file.getCanonicalPath();
+            folderPath = file.getParentFile().getCanonicalPath();
+        }
+        catch (IOException ex) {
+            filePath = file.getAbsolutePath();
```

@rnveach we shouldn't ignore any exceptions unless absolutely necessary. What are cases when absolute path works but canonical path fails?